components/KiboAddressForm.vue

```diff
@@ -137,9 +177,88 @@ export default {
       })
     }
 
+    const errors = ref({
+      firstName: "",
+      lastNameOrSurname: "",
+      address1: "",
+      address2: "",
+      cityOrTown: "",
+      stateOrProvince: "",
+      postalOrZipCode: "",
+      countryCode: "",
+      phoneNumbers: "",
+    })
+
+    const assignErrors = (errorField = "", errorMessage = "") => {
+      if (errorField) {
+        errors.value[errorField] = errorMessage
+      } else {
+        Object.keys(errors.value).filter((field) => (errors.value[field] = errorMessage))
+      }
+    }
+    const validateForm = () => {
+      schema
+        .validate(props.value)
+        .then(() => {
+          context.emit("validateForm", true)
+        })
+        .catch(() => {
+          context.emit("validateForm", false)
+        })
+    }
+    const schema = yup.object({
```

Comment:
can we move schema definition and validate outside of the component?
How can we write unit tests for these validations here?

Reply:
Resolved.